test/ysh-runtime-errors.sh

```diff
@@ -854,6 +854,45 @@ test-method-type-errors() {
    _expr-error-case "var m = 'hi' => leftMatch(/'hi'/); = m => group(3.14)"
 }
 
+test-str-replace() {
+   # Some ad hoc tests - spec tests cover this
+   if false; then
+     _should-run '= "hi" => replace("i", "b")'
+     _should-run '= "hi" => replace(/[a-z]/, "b")'
+     _should-run '= "hi" => replace(/[a-z]/, "b", count=1)'
+     _should-run '= "foo42" => replace(/<capture d+>/, ^"hi $1")'
+     _should-run '= "foo42" => replace(/<capture d+ as num>/, ^"hi $num")'
+     _should-run '= "foo42" => replace(/<capture d+ as num>/, ^"hi ${num}")'
+     _should-run '= "foo42" => replace(/<capture d+ as num>/, ^"hi $[num]")'
+     # test out globals - is this desirable?
+     _should-run '= "foo42" => replace(/<capture d+ as num>/, ^["hi $[num] $PATH"])'
+     # -1 is replace all
+     _should-run '= "foo" => replace("o", "x", count=-1)'
+     _should-run '= "foo" => replace("o", "x", count=-2)'
+   fi
+   # Replace empty string?  Weird Python behavior
+   _should-run '= "foo" => replace("", "-")'
+   _should-run '= "foo" => replace("", "-", count=2)'
+
+   # Use Expr with string
+   _should-run '= "foo" => replace("o", ^"-")'
+   # $0 is regular $0 here
+   _should-run '= "foo" => replace("o", ^"-$0")'
```

Comment:
I just pushed this change noting the difference between

```
replace("o", Expr)  # string to expr
replace(/[o]/, Expr)  # eggex to expr
```

I think there is an argument for disallowing the former, since it probably won't support  `$0`  ??

That is, we could have

```
replace(Str, Str)
replace(Eggex, Str)
replace(Eggex, Expr)
```

but not

```
replace(Str, Expr)
```

Either that or support $0 in the last case.  It doesn't seem to add much power ?